pkg/servicemesh/memberroll_test.go

```diff
@@ -0,0 +1,369 @@
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testMemberRollBuilder.Definition, testMemberRollBuilder.Object)
+		} else {
+			assert.Equal(t, testCase.expectedError, err.Error())
+		}
+	}
+}
+
+func TestMemberRollDelete(t *testing.T) {
+	testCases := []struct {
+		testMemberRoll *MemberRollBuilder
+		expectedError  error
+	}{
+		{
+			testMemberRoll: buildValidMemberRollBuilder(buildMemberRollClientWithDummyObject()),
+			expectedError:  nil,
+		},
+		{
+			testMemberRoll: buildInValidMemberRollBuilder(buildMemberRollClientWithDummyObject()),
+			expectedError:  fmt.Errorf("serviceMeshMemberRoll 'name' cannot be empty"),
+		},
+		{
+			testMemberRoll: buildValidMemberRollBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:  nil,
+		},
+	}
+
+	for _, testCase := range testCases {
+		err := testCase.testMemberRoll.Delete()
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testCase.testMemberRoll.Object)
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		}
+	}
+}
+
+func TestMemberRollUpdate(t *testing.T) {
+	testCases := []struct {
+		testMemberRoll *MemberRollBuilder
+		expectedError  string
+		members        []string
+	}{
+		{
+			testMemberRoll: buildValidMemberRollBuilder(buildMemberRollClientWithDummyObject()),
+			expectedError:  "",
+			members:        newMembersList,
+		},
+		{
+			testMemberRoll: buildInValidMemberRollBuilder(buildMemberRollClientWithDummyObject()),
+			expectedError:  "serviceMeshMemberRoll 'name' cannot be empty",
+			members:        newMembersList,
+		},
+	}
+
+	for _, testCase := range testCases {
+		assert.Equal(t, defaultMembersList, testCase.testMemberRoll.Definition.Spec.Members)
+		assert.Nil(t, nil, testCase.testMemberRoll.Object)
+		testCase.testMemberRoll.WithMembersList(testCase.members)
+		_, err := testCase.testMemberRoll.Update(true)
+
+		if testCase.expectedError != "" {
+			assert.Equal(t, testCase.expectedError, err.Error())
+		} else {
+			assert.Equal(t, testCase.members,
+				testCase.testMemberRoll.Definition.Spec.Members)
+		}
+	}
+}
+
+func TestMemberRollWithMembers(t *testing.T) {
+	testCases := []struct {
+		testMembers       []string
+		expectedError     bool
+		expectedErrorText string
+	}{
+		{
+			testMembers:       newMembersList,
+			expectedError:     false,
+			expectedErrorText: "",
+		},
+		{
+			testMembers:       nil,
+			expectedError:     true,
+			expectedErrorText: "can not modify memberRoll config with empty membersList",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := buildValidMemberRollBuilder(buildMemberRollClientWithDummyObject())
+
+		result := testBuilder.WithMembersList(testCase.testMembers)
+
+		if testCase.expectedError {
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, result.errorMsg)
+			}
+		} else {
+			assert.NotNil(t, result)
+			assert.Equal(t, testCase.testMembers, result.Definition.Spec.Members)
+		}
+	}
+}
+
```

Comment:
Please add TestMemberRollIsReady test case.

Reply:
how can I test the status of the dummy object?

Reply:
Please refer to https://github.com/openshift-kni/eco-goinfra/blob/c6f7a7e7c45e777f1ecf316b87c02ac613ae03d3/pkg/bmh/baremetalhost_test.go#L832